tools/metricq-summary/src/summary.cpp

```diff
@@ -28,13 +28,23 @@
 
 #include "summary.hpp"
 
-Summary Summary::calculate(std::vector<metricq::TimeValue>&& tv_pairs)
+Summary Summary::calculate(std::vector<metricq::TimeValue>&& tv_pairs, std::chrono::milliseconds start_delta, std::chrono::milliseconds stop_delta)
 {
-    assert(tv_pairs.size() > 0);
-
     auto begin = tv_pairs.begin();
     auto end = tv_pairs.end();
 
+    auto start = tv_pairs.front().time;
+    auto stop = tv_pairs.back().time;
+
+    tv_pairs.erase(std::remove_if(begin, end, [&start, stop, start_delta, stop_delta](metricq::TimeValue& tv) {
+        return tv.time <= start + start_delta || tv.time >= stop - stop_delta;
```

Comment:
Because of the last semantic of metric values, this is technically wrong. You'd need one past the end.